src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -159,6 +159,9 @@ public function request(string $method, string $url, array $options = []): Respo
             $this->multi->dnsCache = $options['resolve'] + $this->multi->dnsCache;
         }
 
+        // We keep a reference of the real URL for the logger.
+        $stringUrl = implode('', $url);
```

Comment:
$realUrl?

Reply:
alternatively, what about logging here?

Reply:
I though about it as well but I didn't want to make this change yet until I had some feedback first. If you think it's also valid to log up there, it's fine.

Reply:
Fixed.